src/Core/Manifest/ClassManifest.php

```diff
@@ -496,4 +509,30 @@ protected function coalesceDescendants($class)
             return array();
         }
     }
+
+    /**
+     * Verify that cached data is valid for a single item
+     *
+     * @param array $data
+     * @return bool
+     */
+    protected function validateItemCache($data)
+    {
+        foreach (['classes', 'interfaces', 'traits'] as $key) {
+            // Must be set
+            if (!isset($data[$key])) {
+                return false;
+            }
+            // and an array
+            if (!is_array($data[$key])) {
+                return false;
+            }
+            // No numeric keys
+            $numericKeys = array_filter(array_keys($data), 'is_numeric');
```

Comment:
Summary of offline discussion: @tractorcow will update this code to only check the first element of `$data`, which will detect and reject old cache content much more quickly. A comment will be added to note that the purpose of this is to detect out-of-date caches.